### DISCUSSION TITLE
Ignore SSL certificates when fetching GraphQL data

### DISCUSSION THREAD

**Discussion by maxyudin** [vercel/next.js#26296] [Help]
I am on the localhost with *.test domains for each project. Following [the guide by Cecile Muller](https://gist.github.com/cecilemuller/9492b848eb8fe46d462abeb26656c4f8) I've created self-issued certificates for testing purposes. Now I'm experimenting with Next.js and facing the problem when I can't fetch the GraphQL output from my localhost-based WordPress because of "_FetchError: request to XXXXX failed, reason: unable to verify the first certificate_". Errno and error code are "UNABLE_TO_VERIFY_LEAF_SIGNATURE". All my browsers  eats this certificate like their own food, but not _node-fetch_ and others from NodeJS.

How can I ignore certificate verification when fetching the data? I can't see any CLI option or config string to do this like, e. g.,  _curl_ can do. Should I write (copy/paste) a script to include it somewhere? 

[ssl-root-cas](https://www.npmjs.com/package/ssl-root-cas) looks like a solution, but I can't figure out how to nail it to the brand-new starter project.

**Comment by kmgdevelopment** [comment]
Assuming you local Wordpress install is available over SSL, you can add `--experimental-https` after the Next.js dev server command in `package.json`:

```
"scripts": {
    "dev": "next dev --experimental-https",
    "build": "next build",
    "start": "next start",
    "lint": "next lint"
},
```

[More info in the Next.js docs](https://nextjs.org/docs/app/api-reference/cli/next#using-https-during-development)

**Comment by benjdiasaad** [comment]
Hi @maxyudin,

This looks more like a Node.js TLS issue rather than something related to the Next.js dev server itself.

The error:

"UNABLE_TO_VERIFY_LEAF_SIGNATURE"

usually happens because Node.js does not trust self-signed certificates by default, even if your browser does.

So enabling HTTPS for the Next.js dev server (`--experimental-https`) won’t fix the root cause here, since the failure occurs during the fetch request.

A few approaches that typically work:

- temporarily disable TLS verification (for local development only):
  
  ```bash
  NODE_TLS_REJECT_UNAUTHORIZED=0
